src/index.js

```diff
@@ -686,14 +686,59 @@ PubSub.prototype.request = function(config, callback) {
 
   if (!gaxClient) {
     // Lazily instantiate client.
-    gaxClient = v1(this.options)[config.client](this.options);
+    if (config.client === 'PublisherClient') {
+      gaxClient = new v1.PublisherClient(this.options);
+    } else if (config.client === 'SubscriberClient') {
+      gaxClient = new v1.SubscriberClient(this.options);
+    } else {
+      let err = new Error('Client is unknown: ' + config.client);
+      callback(err);
+    }
+
+    // Determine what scopes are needed.
+    // It is the union of the scopes on all three clients.
+    let clientClasses = [v1.SubscriberClient, v1.PublisherClient];
+
+    let allScopes = {};
+    for (let clientClass of clientClasses) {
+      for (let scope of clientClass.scopes) {
+        allScopes[scope] = true;
+      }
+    }
+
+    var scopes = Object.keys(allScopes);
+    this.options.scopes = scopes;
```

Comment:
I don't believe we have a test for this functionality (getting the scopes, then assigning them to `option`)

Reply:
It's verified independently by system tests, but I agree unit test is needed as well. Will add.